### ISSUE TITLE
bug: Ion-modal getting close on reopening

### ISSUE THREAD

**Issue by Saqib92** [ionic-team/ionic-framework#24725]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

inline ion-modal getting auto close when reopen, Modal open first time correctly. then opening it again suddenly close it. 

### Expected Behavior

open modal again should stay opened. 

### Steps to Reproduce

Modal Open Html:
```
<ion-button (click)="openModal()">Open Modal</ion-button>
```

Modal HTML:
```
<ion-modal [isOpen]="open">
  <ng-template>
    <ion-content>
        Modal Content
       <ion-button (click)="openModal()">Close Modal<ion-button>
    </ion-content>
  </ng-template>
</ion-modal>
```

.ts
```

open: boolean = false;

openModal(){
   this.open = !this.open;
}
```

### Code Reproduction URL

[https://github.com/Saqib92/Multiple-inline-modal-issue](https://github.com/Saqib92/Multiple-inline-modal-issue)

### Ionic Info

```
Ionic:

   Ionic CLI                     : 6.18.1 (C:\Users\IDEA\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 6.0.5
   @angular-devkit/build-angular : 13.0.4
   @angular-devkit/schematics    : 13.0.4
   @angular/cli                  : 13.0.4
   @ionic/angular-toolkit        : 5.0.3

Capacitor:

   Capacitor CLI      : 3.4.0
   @capacitor/android : not installed
   @capacitor/core    : 3.4.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.3
   native-run  : 1.5.0

System:

   NodeJS : v16.13.1 (C:\Program Files\nodejs\node.exe)
   npm    : 8.3.2
   OS     : Windows 10
```

### Additional Information

Latest Ionic 6 using Ion modal 2nd time closes it self.

**Comment by Saqib92** [comment]
Temporary fix using `setTimeout()`

```
openModal(){
  setTimeout(()=>{
     this.open = !this.open;
  }) 
}
```

**Comment by Saqib92** [comment]
do we have ionic 6 Stackblitz?

**Comment by Saqib92** [comment]
It's Working now in Demo project i created for Testing. But in my Actual Project having Same Dependencies and Code its getting close on 2nd opening. Will Investigate it again. Closing for now.

**Comment by Saqib92** [comment]
Found Issue:
Having Multiple Inline Modals Causing this behaviour: 
Try opening both on input click Randomly and Alternatively. 

Here is Git Report for MVP:
https://github.com/Saqib92/Multiple-inline-modal-issue

**Comment by liamdebeasi** [comment]
Hey there,

I am getting an error when trying to run your sample app:

```
ng] - Generating browser application bundles (phase: setup)...
[ng] file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:4616
[ng]       throw new Error(`The target entry-point "${invalidTarget.entryPoint.name}" has missing dependencies:
[ng]             ^
[ng] Error: The target entry-point "ion2-calendar" has missing dependencies:
[ng]  - moment
[ng]     at TargetedEntryPointFinder.findEntryPoints (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:4616:13)
[ng]     at file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:5008:33
[ng]     at SingleProcessExecutorSync.doExecute (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:14633:23)
[ng]     at file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:14654:35
[ng]     at SyncLocker.lock (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:14884:14)
[ng]     at SingleProcessExecutorSync.execute (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:14654:19)
[ng]     at mainNgcc (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:15985:19)
[ng]     at Module.process2 (file:///Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@angular/compiler-cli/bundles/ngcc/index.js:16050:10)
[ng]     at NgccProcessor.processModule (/Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@ngtools/webpack/src/ngcc_processor.js:182:27)
[ng]     at /Users/liamdebeasi/Ionic/Multiple-inline-modal-issue/node_modules/@ngtools/webpack/src/ivy/host.js:146:18
```

Can you double check your project?

**Comment by Saqib92** [comment]
run `npm i moment` for testing i added multiple dependencies because of my actual project.

**Comment by Saqib92** [comment]
open chrome  / Edge dev tool change device to any Iphone  device ideally to `6/7/8 plus` open 1 first modal twice and then 2nd modal.

**Comment by liamdebeasi** [comment]
Thanks! It appears this issue has already been fixed in Ionic 6.0.6, so I am going to close this.

**Comment by Saqib92** [comment]
yes, Not happening  after upgrading to `v6.0.6`
